src/bin/pomodoro.rs

```diff
@@ -0,0 +1,300 @@
+                    tomtom.remaining_time = tomtom.toma_time;
+                };
+            } else {
+                tomtom.running = true;
+                update_button(cb_button);
+                if tomtom.break_period {
+                    tomtom.remaining_time = tomtom.break_time;
+                } else {
+                    tomtom.remaining_time = tomtom.toma_time;
+                };
+                update_timer(&mut tomtom);
+                add_timeout_countdown(tomaty.clone());
+            };
+            println!("Button clicked!");
+        });
+
+    }
+
+    fn alarm() {
+        println!("POMODORO OVER, TAKE A BREAK!");
+    }
+
+    fn add_timeout_countdown(tomaty: Rc<RefCell<Tomaty>>) {
+        gtk::timeout_add_seconds(1, move || {
+            let mut tomtom = tomaty.borrow_mut();
+            if tomtom.remaining_time == Duration::seconds(0) {
+                alarm();
+                tomtom.running = false;
+                update_button(&tomtom.tomaty_button);
+                if tomtom.break_period {
+                    tomtom.timer_label.set_markup(BREAK_MSG);
+                    tomtom.break_period = false;
+                } else {
+                    tomtom.tomatos_completed += 1;
+                    let count_formatted =
+                        format!(COUNT!(), tomtom.tomatos_completed);
+                    tomtom.count_label.set_markup(&count_formatted);
+                    let total = tomtom.tomatoro_length * tomtom.tomatos_completed;
+                    let total_formatted =
+                        format!(TOTAL_TIME!(), total.num_minutes());
+                    tomtom.total_label.set_markup(&total_formatted);
+                    tomtom.timer_label.set_markup(TOMA_MSG);
+                    tomtom.break_period = true;
+                }
+                return gtk::Continue(false)
+            }
+            if !tomtom.running {
+                return gtk::Continue(false)
+            }
+            tomtom.remaining_time = tomtom.remaining_time - Duration::seconds(1);
+            update_timer(&mut tomtom);
+            gtk::Continue(true)
+        });
+    }
+
+    fn update_button(button: &gtk::Button) {
+        match button.get_label().as_ref().map(String::as_ref) {
+            Some("start") => button.set_label("restart"),
+            _ => button.set_label("start"),
+        }
+    }
+
+    fn build_ui(application: &gtk::Application) {
+        // let window = gtk::Window::new(gtk::WindowType::Toplevel);
+        let window = gtk::ApplicationWindow::new(application);
+        window.set_application(application);
+        window.set_title("tomaty: gtk::Focus");
+        window.set_border_width(5);
+        window.set_resizable(false);
+
+        window.set_position(gtk::WindowPosition::Center);
+        window.set_default_size(350, 70);
+
+        window.connect_delete_event(clone!(window => move|_, _| {
+            window.destroy();
```

Comment:
Minor problem with `clone` macro: now it produce warning if build without features.
Please or move `macro_rules!` inside `mod pomodoro` 
or remove it and change this to
```
let window_clone = window.clone();
window.connect_delete_event(move |_, _| {
    window_clone.destroy();
```
as it single macro usage